src/ddl_utils.c

```diff
@@ -0,0 +1,72 @@
+#include <postgres.h>
+#include <tcop/deparse_utility.h>
+
+/*
+ * Test if ddl command as an alter table add column
+ */
+PG_FUNCTION_INFO_V1(ddl_is_add_column);
+Datum
+ddl_is_add_column(PG_FUNCTION_ARGS)
```

Comment:
I started to do something like that but it grew so I opted to only fix the known bug in this PR. I do think we should move more of this to c as we move along though but I don't think it is the  right prio refactoring stuff that works right now.